src/tests/GfxBatchRendererTest.cpp

```diff
@@ -2156,6 +2175,83 @@ void GfxBatchRendererTest::imageInto() {
   CORRADE_COMPARE(depth.pixels<Mn::Float>()[64][48], 0.0909091f);
 }
 
+void GfxBatchRendererTest::depthUnprojection() {
+  constexpr int tileCountX = 2;
+  constexpr int tileCountY = 2;
+  constexpr int envCount = tileCountX * tileCountY;
+  constexpr float near = 0.001f;
+  constexpr float far = 10.0f;
+
+  // clang-format off
+  esp::gfx_batch::RendererStandalone renderer{
+      esp::gfx_batch::RendererConfiguration{}
+          .setTileSizeCount({64, 64}, {tileCountX, tileCountY}),
+      esp::gfx_batch::RendererStandaloneConfiguration{}
+          .setFlags(esp::gfx_batch::RendererStandaloneFlag::QuietLog)
+  };
+  // clang-format on
+
+  CORRADE_VERIFY(renderer.addFile(
+      Cr::Utility::Path::join(TEST_ASSETS, "scenes/batch.gltf")));
+
+  // Place environment cameras at various distances from origin.
+  const auto& projection =
+      Mn::Matrix4::perspectiveProjection(60.0_degf, 1.0f, near, far);
+  renderer.updateCamera(
+      0, projection,
+      Mn::Matrix4::translation(Mn::Vector3::zAxis(2.5f)).inverted());
+  renderer.updateCamera(
+      1, projection,
+      Mn::Matrix4::translation(Mn::Vector3::zAxis(5.0f)).inverted());
+  renderer.updateCamera(
+      2, projection,
+      Mn::Matrix4::translation(Mn::Vector3::zAxis(7.5f)).inverted());
+  renderer.updateCamera(
+      3, projection,
+      Mn::Matrix4::translation(Mn::Vector3::zAxis(20.0f)).inverted());
+
+  // Spawn a plane in each environment, at origin, facing the camera.
+  CORRADE_VERIFY(renderer.hasNodeHierarchy("square"));
+  for (int i = 0; i < envCount; ++i) {
+    CORRADE_COMPARE(renderer.addNodeHierarchy(i, "square"), 0);
+  }
+
+  // Render.
+  renderer.draw();
+  Mn::Image2D depth = renderer.depthImage();
+  MAGNUM_VERIFY_NO_GL_ERROR();
+
+  // Unproject each scene.
+  Mn::MutableImageView2D depthBufferView{Mn::PixelFormat::Depth32F,
+                                         depth.size(), depth.data()};
+  const size_t subBufferDataSize = depth.data().size() / envCount;
+  const auto subBufferSize = Mn::VectorTypeFor<2, int>{
+      depth.size().x() / tileCountX, depth.size().y() / tileCountY};
```

Comment:
`Mn::VectorTypeFor<2, int>` is simply a `Mn::Vector2i`. Code-golfing further, having tileCount as a `Vector2i` as well, it'll be just

```cpp
Mn::Vector2i subBufferSize = depth.size()/tileCount;
```

Or you can save that `{64, 64}` from the configuration into a constant and use it here :)